service/controller/app/v1/resource/chart/desired.go

```diff
@@ -0,0 +1,63 @@
+package chart
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	catalogName := key.CatalogName(cr)
+
+	appCatalog, err := r.g8sClient.ApplicationV1alpha1().AppCatalogs("default").Get(catalogName, v1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return nil, microerror.Maskf(notFoundError, "appCatalog %#q in namespace %#q", catalogName, "default")
+	} else if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	chartURL := generateCatalogURL(appCatalog.Spec.CatalogStorage.URL, cr.Spec.Name, cr.Spec.Release)
+
+	chartCR := &v1alpha1.Chart{
+		TypeMeta: v1.TypeMeta{
+			Kind:       chartKind,
+			APIVersion: chartAPIVersion,
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:        cr.Spec.Name,
+			Labels:      cr.GetObjectMeta().GetLabels(),
+			Annotations: cr.GetObjectMeta().GetAnnotations(),
+		},
+		Spec: v1alpha1.ChartSpec{
+			Name:       cr.GetObjectMeta().GetName(),
+			Namespace:  cr.Spec.Namespace,
+			TarballURL: chartURL,
+		},
+	}
+
+	chartCR.Spec.KubeConfig.Secret.Name = cr.Spec.KubeConfig.Secret.Name
+	chartCR.Spec.KubeConfig.Secret.Namespace = cr.Spec.KubeConfig.Secret.Namespace
```

Comment:
Lets remove the setting of the KubeConfig.

The app CRs have the kubeconfig set because they need to be created in the tenant cluster. The chart CRs are managed by chart-operator in the same cluster. So we don't need to set the kubeconfig.

Reply:
@rossf7 If that is the case, then we should delete this `kubeConfig` part in `chart` CR specification. https://github.com/giantswarm/giantswarm/blob/b6af8e15df1600aada45567fbd37e053d1403e1f/roadmap/MANAGED-SERVICES-CATALOG.MD#release-process